pkg/apis/kops/validation/validation.go

```diff
@@ -675,7 +675,19 @@ func validateKubelet(k *kops.KubeletConfigSpec, c *kops.Cluster, kubeletPath *fi
 				allErrs = append(allErrs, field.Forbidden(kubeletPath.Child("cpuCFSQuotaPeriod"), "cpuCFSQuotaPeriod has been removed on Kubernetes >=1.20"))
 			}
 		}
-
+		if k.ShutdownGracePeriod != nil {
+			if k.ShutdownGracePeriod.Seconds() > 30 {
+				allErrs = append(allErrs, field.Invalid(kubeletPath.Child("shutdownGracePeriod"), k.ShutdownGracePeriod.String(), "shutdownGracePeriod cannot exceed 30 seconds"))
```

Comment:
Where is the reference for this limitation?

Reply:
This is the max amount of seconds a shutdown may be inhibited. I don't have any reference for this,but setting a higher value will fail.

Reply:
Where/how does it fail? This is really unfortunate; stateful applications should be able to inhibit this for hours while they increase the replication of the node-local data they're about to lose.

Reply:
Kubelet errorred out with a message that it was too high and refused to set the inhibitor. Not sure where the error came from kubelet or systemd. I will reproduce and have a closer look.

Unattended upgrades holds an inhibitor lock for much longer than 30s when configured to upgrade on shutdown.

Reply:
```
Dec 22 07:46:49 ip-172-20-59-237 kubelet[5072]: E1222 07:46:49.647969    5072 kubelet.go:1443] "Failed to start node shutdown manager" err="node shutdown manager was unable to update logind InhibitDelayMaxSec to 1m0s (ShutdownGracePeriod), current value of InhibitDelayMaxSec (30s) is less than requested ShutdownGracePeriod"
```

So it is a logind config. I saw this one before, but didn't think of it much because the default is 5s, not 30s.
But turns out ubuntu has this:
```
# cat /usr/lib/systemd/logind.conf.d/unattended-upgrades-logind-maxdelay.conf 
[Login]
# delay
InhibitDelayMaxSec=30
```

We could drop in a file as well. But I suggest doing that in a followup.

Reply:
For reference regarding filenames:
>  Files in the *.conf.d/ configuration subdirectories are sorted by their filename in lexicographic order, regardless of in which of the subdirectories they reside. When multiple files specify the same option, for options which accept just a single value, the entry in the file sorted last takes precedence (..)

So yay to unattended-upgrades for starting their file name at "u"

Reply:
Does Debian 11 have the same limitation? If not, we should remove this check and just document it.

Reply:
I would assume unattened-upgrades install the same limit increase. If it doesn't, then the limit is 5s and the inhibitor lock won't be installed.

Reply:
I still think such a validation is not very reliable for multiple OSes and documenting the limitation until we override would be preferable. Operators could drop an override by other means until then.

Reply:
Sounds reasonable.